syntax = "proto3";

message Entry {
  int32 term = 1;   // Term when entry was received by leader
  int32 index = 2;   // Position of entry in the log
  string command = 3; // Command for state machine
  int32 id = 4; // Transaction id
}

message AppendEntriesReq {
  int32 term = 1; // Leader's term
  int32 leaderId = 2; // So follower can redirect clients
  int32 prevLogIndex = 3; // Index of log entry immediately preceding new ones
  int32 prevLogTerm = 4; // Term of prevLogIndex entry
  repeated Entry entries = 5; // Log entries to store (empty for heartbeat)
  int32 commitIndex = 6; // Last entry known to be committed
  int32 serverId = 7; // Receiver of this request
}